docs/attachapi.md

```diff
@@ -0,0 +1,99 @@
+security, only the process owner can connect to their processes.
+
+On UNIX systems, the Attach API uses `/tmp` and creates a common subdirectory, which is `.com_ibm_tools_attach` by default.
+Security is handled by POSIX file permissions. The Attach API directory must be owned by `root` user and must have read, write, and execute file permissioms for
+`user`, `group`, and `other` (drwxrwxrwx). The sticky bit must be set so that only the owner and `root` can delete or rename files or directories within it.
+
+```
+~/tmp $ ls -al
+total 0
+drwxr-xr-x   3 user_a staff    96  6 Aug 17:11 .
+drwxr-xr-x+ 89 user_a staff  2848  6 Aug 17:11 ..
+drwxrwxrwx+  7 root   staff   224  6 Aug 17:22 .com_ibm_tools_attach
+```
+
+In the default Attach API directory you can find certain files that start with an underscore `_*`, which are involved in synchronization.
+These files can be owned by any user but must have read and write permissions set. The files are empty and are automatically re-created if deleted.
+When your application attaches to a VM, a process directory is created.
+
+```
+~/tmp/.com_ibm_tools_attach $ ls -l
+total 3
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _attach_lock
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _master
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _notifier
+drwx--x--x  6 user_b  staff  192  6 Aug 17:21 process_a
+```
+
+The files in the subdirectory for a process, with the exception of a lock file, are accessible only by the owner of a process. The permissions
+for these files are `rwxr-xr-x` with the exception of the `attachNotificationSync` file, as shown in the following example.
+
+```
+~/tmp/.com_ibm_tools_attach/process_a $ ls -l
+total 4
+-rwxrw-rw-  1 user_b  staff  0  6 Aug 17:18 attachNotificationSync
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_a
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_b
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_c
+```
+
+**Notes on z/OS:**
+
+- z/OS systems must also set POSIX permissions on files and cannot rely on RACF&reg; or system level security to protect applications.
+- To avoid z/OS console messages reporting security violations in `/tmp`, add a security exception or specify a different common subdirectory by setting the `com.ibm.tools.attach.directory` system property.
+
+## Configuring the Attach API
+
+A number of system properties are available to configure the Attach API when you start a Java application, as shown in the following table:
+
+| System property                                                                             |    Description                                                           |
+|---------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| [`-Dcom.ibm.tools.attach.directory=<directory_name>`](comibmtoolsattachdirectory.md)        | Specify a different directory in which to create `.com_ibm_tools_attach` |
```

Comment:
I know this probably comes from an old internal doc during our slack conversation.
`Specify a different common directory for Attach API working files.` is correct, the specified directory is going to be the Attach API common subdirectory, i.e., replace `(new File(systemTmpDir,".com_ibm_tools_attach")).getPath()`.
Please remove this modification.

Reply:
Changed back. Also applied to the system property topic.